MagicSpellKeyboard/LetterMapping.swift

```diff
@@ -33,7 +33,45 @@ class LetterMapping {
         [Finger(side: .Right, name: .Index), Finger(side: .Right, name: .Middle)] : "z",
     ]
     
-    static func getLetter(fingers: Set<Finger>) -> String? {
-        return map[fingers]
+    private static let upperCaseMap: [Set<Finger>: String] = [
```

Comment:
This is not a good way to implement this. You will need to maintain two data sets instead of just one. Use .uppercaseString instead.